webapp/components/Registration/CreateUserAccount.vue

```diff
@@ -84,9 +87,24 @@ import gql from 'graphql-tag'
 import PasswordStrength from '../Password/Strength'
 import { SweetalertIcon } from 'vue-sweetalert-icons'
 import PasswordForm from '~/components/utils/PasswordFormHelper'
+import { VERSION } from '~/constants/terms-and-conditions-version.js'
+
+/* TODO: hier muss die version rein */
 export const SignupVerificationMutation = gql`
-  mutation($nonce: String!, $name: String!, $email: String!, $password: String!) {
-    SignupVerification(nonce: $nonce, email: $email, name: $name, password: $password) {
+  mutation(
+    $nonce: String!
+    $name: String!
+    $email: String!
+    $password: String!
+    $termsAndConditionsAgreedVersion: String!
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-26T07:52:26Z" title="Monday, August 26th 2019, 9:52:26 am +02:00">Aug 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -84,9 +87,24 @@ import gql from 'graphql-tag'
 import PasswordStrength from '../Password/Strength'
 import { SweetalertIcon } from 'vue-sweetalert-icons'
 import PasswordForm from '~/components/utils/PasswordFormHelper'
+import { VERSION } from '~/pages/terms-and-conditions'
+
+/* TODO: hier muss die version rein */
 export const SignupVerificationMutation = gql`
-  mutation($nonce: String!, $name: String!, $email: String!, $password: String!) {
-    SignupVerification(nonce: $nonce, email: $email, name: $name, password: $password) {
+  mutation(
+    $nonce: String!
+    $name: String!
+    $email: String!
+    $password: String!
+    $termsAndConditionsAgreedVersion: String!
```


Has the exclamation mark to be removed here as well?!
… after @roschaefer suggestion?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-30T11:12:59Z" title="Friday, August 30th 2019, 1:12:59 pm +02:00">Aug 30, 2019</time>_

---


This is an **input** param! So if the user creates the account, he/she has to tell us which version of the terms and conditions he/she agreed to. This is *different* from the scenario, that we import the database (version is `undefined`) and the server should respond with an object of type `User` and unfortunately that is a user who has not agreed to any terms and conditions. If the server would respond `undefined` for `User.termsAndConditions` the server would crash.

Does that make sense? @Tirokk So the exclamation mark here is correct.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-30T11:14:38Z" title="Friday, August 30th 2019, 1:14:38 pm +02:00">Aug 30, 2019</time>_

---


Actually, the server will still crash. Ie. if you request `termsAndConditionsAgreedVersion` and we haven't used `neo4jgraphqljs` in the resolver. It is a rare situation though.

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-30T11:54:54Z" title="Friday, August 30th 2019, 1:54:54 pm +02:00">Aug 30, 2019</time>_

---


Yes, that makes absolutely sense for me now @roschaefer .
I got it already as I was closer on the code with @ogerly .